v3.0

stovak opened the pull request:
• Update to dependency libraries
• Homebrew installation
• D9ify and Local development commands

uberhacker committed:
Add PluginCommand to install, list, uninstall and update plugins

Commit:
Update to fix code sniffer issues

uberhacker committed:
Add ability to install plugins via 'composer create-project'

Commit:
Refactored plugin commands into separate class files

Commit:
Update to fix TODO notes

  - Change messages to be more consistent
  - Clean up docblocks

Commit:
Fix to get the major version based on the installed Terminus release

  - Backup plugin projects that do not have a .git directory during updates

Commit:
Add the 'Terminus major version' variable in all relevant locations

  - Remove archive installs from updates
  - Change project backup location

uberhacker committed:
Add requirements check

  - Remove obsolete or unused methods
  - Provide composer only install option

uberhacker committed:
Refactor requirements check

uberhacker committed:
Refactor requirements check

  - Reformat search results output

Commit:
Add empty project check with usage text to install command

Commit:
Fix coding standards issue with SearchCommand.php

greg-1-anderson committed:
Touch-up plugin:search command.

Commit:
Rename $approved to $approvedProjects in SearchCommand.php line 76

Commit:
Add \@param to docblock of checkStatus method in SearchCommand.php

Commit:
Remove unreferenced use line in SearchCommand.php

Commit:
Bypass Symfony exception and display something more understandable when there is no keyword supplied with the plugin:search command

Commit:
Add stability option to InstallCommand.php

Commit:
Fix @option stability in InstallCommand.php

greg-1-anderson committed:
Register pancakes as an 'Approved' plugin. Rename 'Unknown' to 'Unofficial'.

uberhacker committed:
Update for better Windows support

  - Move updatePlugin method from PluginBaseCommand.php to
  UpdateCommand.php
  - Remove getSlash method in PluginBaseCommand.php

uberhacker committed:
Remove PluginCommand.php

uberhacker committed:
Add DIRECTORY_SEPARATOR to the getPluginProjects method in PluginBaseCommand.php

uberhacker committed:
Add smarter sorting of version tags, search and list results

uberhacker committed:
Update to the latest stable release if a previous non-stable semantic version is available

uberhacker committed:
Add semverCompare method to correctly compare semantic versions

uberhacker committed:
Replace semverCompare method with Composer\Semver\Comparator

iamEAP committed:
Fix plugin updates on OS X.

iamEAP committed:
Adding functional tests for plugin management.

SaraMcCutcheon committed:
Line length linter fixes

SaraMcCutcheon committed:
Moved plugin commands into self namespace to avoid confusion with WP(-CLI) plugins

SaraMcCutcheon committed:
Prefer brakets for arrays

SaraMcCutcheon committed:
Updated to reflect prev change

SaraMcCutcheon committed:
Updated CHANGELOG with new plugin commands

SaraMcCutcheon committed:
Updated README

SaraMcCutcheon committed:
Refactored self:plugin:install

SaraMcCutcheon committed:
Refactored self:plugin:list to use PluginDiscovery/PluginInfo

SaraMcCutcheon committed:
Refactored self:plugin:search

SaraMcCutcheon committed:
Refactored the self:plugin:install command

SaraMcCutcheon committed:
Started refactoring the self:plugin:update command

SaraMcCutcheon committed:
More refactory

SaraMcCutcheon committed:
More refactoring on plugin base command

SaraMcCutcheon committed:
Refactory

SaraMcCutcheon committed:
More refactoring

uberhacker committed:
Fix php parse errors (#2059)

yuri-pochepko committed:
Fixed typo for an example in comments (#2103)

Clarify remote (not local) backup. (#2106)

Per https://github.com/pantheon-systems/documentation/issues/6137, customers may try to use this command to restore from a local backup file. Clarification is being provided in the docs via https://github.com/pantheon-systems/documentation/pull/6162, but this brings the clarification right into the source material.

Update README for Installation methods. (#2058)

* document installation of terminus.phar

* note lack of terminus update in other methods

* document when to use each installation method

Catch the 'Not found' error in the Site model getFeature() method (#2065)

[PLAT-1642] Connection usernames are derived from data rather than hard-coded (#2107)

* Connection usernames are derived from data rather than hard-coded

* Fixed tests for Environment model

* Fixing mysterious new fail on every Travis PHP 7.3 run (not just this PR)

* Make example also not use static DB username

Updating to version 2.5.0 (#2114)

Updated to 2.5.1-dev (#2115)

Updated forgotten fix in CHANGELOG (#2116)

README.md URLS and Requirements (#2130)

* Documentation Update

 * Updated URL so it always points to the latest version of Terminus
 * Corrected paths so they match path in step 1 of self-contained
   Terminus install

* Updated Requirements and Links

Fixes #2129 - Missing Requirements php-json

* Added `php-json` extension as a requirement
* Updated URLs to allow the user-agent to pick the PHP manual's language

[OTTO-1109] Added ability to change the TTL of automatic backups (#2133)

Upgrading to 2.6.0 (#2137)

Dev version bump/update release date (#2139)

D9ify Command Buildout

Fixing issues with updated libraries as well as adding d9ify command

liting

fixing container invocations

container issues

Container Syntax Updates

working but tests dont pass

Nearing Alpha

dangersly close

WIP

site tests pass

Callin it: Alpha 1

3.0-alpha1

fixing build issues

build issues

fix composer build issue

new build workflow

change artifcat testing name

build update

build issues

d9ify first phase is completely ported and working

d9ify changes

commit-and-push command buildout

commit and push command moved to local group

update workflowProcessing trait

updating github actions build

Process Command and changing functional test

adding homebrew forumula and ssh command fixes

fixing uniquid namespacing

syntax errors

move precommit out of post-update and post-install

fixing issues with homebrew install

testing updates

more functional tests and coverage report

updating coverage generation

test coverage update

moving CI to v7.4

move outdated README out of root directory

Short testing sequence passes

adding composer install

run testing on specific container

build fix

adding pcov

remove container

create tokens dir if not exists

make cache/tokens directory before access

fixing bootstrap for testing

build update

fix build

fixing issue with token/login

moving login to helper trait

env vars

env vars

env values in build

removing debugging

moving to fixed site value rather than random one

fix issue with test failure

fixing process timeout

partial fix #2124

squashed commits: v3.0 library update and buildout

Use --machine-token instead of --token.

Add tmate debugging.

Remove tmate debugging.

Invoke t3 with php.

Localdev Templates

building out package/bottle creation

removing binary. t3 should run under its own power

Add stderr to strings returned from exec

feat: support providing custom ssh command via `ssh_command` option

temporarily removing ubuntu from testing

fixing testing errors

Building out package/bottle builds

updating composer install

tag all builds as pre-release and release manually

functional:long testing

SaraMcCutcheon commented on config/constants.yml:
```diff
@@ -34,6 +34,7 @@ TERMINUS_PLUGINS_DIR:       '[[ TERMINUS_USER_HOME ]]/.terminus/plugins'
 TERMINUS_COMMAND_CACHE_DIR: '[[ TERMINUS_CACHE_DIR ]]/commands'
 TERMINUS_TOKENS_DIR:        '[[ TERMINUS_CACHE_DIR ]]/tokens'
 TERMINUS_ASSETS_DIR:        '[[ TERMINUS_ROOT ]]/assets'
+TERMINUS_LOCAL_COPIES:      '[[ TERMINUS_USER_HOME ]]/pantheon-local-copies'
```
We don't want this in the user's `~` dir.

stovak replied:
Why is that? For local development, I just was looking for a centralized place for any pantheon utility to put site clones. Is your objection to the place or the name?

SaraMcCutcheon commented on src/Collections/SavedTokens.php:
```diff
@@ -44,10 +44,13 @@ public function add($model_data, array $options = [])
-            SavedToken::class,
-            [(object)['token' => $token_string,], ['collection' => $this,]]
-        );
+        $token_nickname = "token-" . \uniqid();
```
I don't seem to have this `uniqid` function. PHP version 7.4.19

SaraMcCutcheon replied:
Possibly a namespacing thing (see `site:list`)
`PHP Fatal error:  Uncaught Error: Call to undefined function Pantheon\Terminus\Models\uniquid() in <TERMINUS>/src/Models/OrganizationSiteMembership.php:60`

stovak replied:
Name spacing/misspelling Fixed.

SaraMcCutcheon reviewed (commented):
Current comments

stovak committed:
test updates

testing update

changing name of LONG test to ALL for clarity

fixing bad test

fixing test

buildout of the release process

adding missing constants

updating build and bundler tools

build changes

adding tests and fixing issue surfaced by tests

linting

error in CI file

linting

testing update

tests, tests and more tests

more tests passing

stovak committed:
fixing coverage template

stovak committed:
updating testing

stovak committed:
updating behat tests and gh actions

stovak committed:
updating test for builds

stovak committed:
Github workflows update and security update to composer packages

stovak committed:
Updating tests and local commands

stovak committed:
upstream updates tests

stovak committed:
fixing upstream tests

kporras07 committed:
Add -S param to env command.

kporras07 committed:
Add missing php7.4-xml dependency.

kporras07 committed:
Set right architecture for the generated package.

kporras07 committed:
Fix package_linux job and create new publish_linux job.

kporras07 committed:
Make behat_tests job run.

stovak committed:
Merge pull request #2153 from kporras07/v3.0-packagecloud

Fix deb package creation and publishing.

stovak committed:
Merge pull request #2152 from kporras07/v3.0-linux-bin

Add -S param to env command.

stovak committed:
Merge pull request #2154 from kporras07/v3.0-behat

Make behat_tests job run.

stovak committed:
building out dockerize command

stovak committed:
pull database/files command

stovak committed:
updating localdev templates

weaver299 committed:
Correct a couple comments

stovak committed:
updateing localdev templates

stovak committed:
Merge pull request #2155 from pantheon-systems/v3.0-test-comments

Correct a couple comments

stovak committed:
updated localdev templates

weaver299 committed:
Domain commands test: primary add/remove tests

Commit:
Add getSiteName() wrapper method for the test site name

Commit:
Fix code style issues in TerminusTestTrait and LoginHelperTrait

Commit:
Add Redis enable/disable commands tests

Commit:
Add Solr enable/disable commands tests

Commit:
Add functional test for import:database command

Commit:
Add functional test for import:files and import:complete commands

Commit:
Merge pull request #2157 from pantheon-systems/v3.0-functional-tests-redis-solr

v3.0 Redis/Solr functional tests

Commit:
Add test placeholder for import:site command

weaver299 committed:
CMS-156: Add TagCommandsTest

stovak committed:
Merge pull request #2161 from pantheon-systems/CMS-155

CMS-155: Add/update domain commands tests

stovak committed:
Merge pull request #2160 from pantheon-systems/CMS-156

CMS-156: Add tag command tests

weaver299 committed:
Replace '@gropu' with '@group'

weaver299 committed:
Fix: new relic test also disables

weaver299 committed:
Merge branch 'newrelictests' into fix_gropu

weaver299 committed:
New relic test takes 7-8 minutes, long group

namespacebrian committed:
Merge pull request #2156 from pantheon-systems/fix_gropu

Correct tests 'gropu' typo to 'group'

weaver299 committed:
Merge branch 'v3.0' into v3.0-functional-tests-import

weaver299 committed:
Fix: tag:list command returning integers instead of tag names

weaver299 committed:
Add TERMINUS_ORG env var to functional tests

namespacebrian committed:
Merge pull request #2158 from pantheon-systems/v3.0-functional-tests-import

v3.0 Functional tests for import:* commands

Commit:
Merge pull request #2162 from pantheon-systems/v3.0-fix-tag-list

Fix tag:list bug

Commit:
Add functional test for "backup:get" to return the most recent backup URL

Commit:
Use vendor/bin/phpcs in composer scripts

Commit:
Add "Validate Code" step to CI "Checkout & build Phar" phase

namespacebrian committed:
Merge pull request #2163: Verify backup:get returns most recent backup

Add functional test for backup:get to verify the most recent backup

namespacebrian committed:
Merge pull request #2164: phpcs config changes

v3.0 phpcs config changes

kporras07 committed:
Merge branch 'v3.0' of github.com:pantheon-systems/terminus into plugin-commands-v3

kporras07 committed:
Initial changes to make it work on t3.

kporras07 committed:
Process arguments should be separate array indexes.

Commit:
Add Lock commands functional tests

Commit:
Fix empty ID and EMAIL attributes of org:people:list result

Commit:
Update functional test for org:people:list command to have better check of ID/EMAIL attributes

stovak committed:
Merge pull request #2165 from pantheon-systems/v3.0-functional-tests-lock-commands

v3.0 Add Lock commands functional tests

kporras07 committed:
Fix getProcess function the right way.

kporras07 committed:
Fix usage of symfony/process to use fromShellCommandline.

kporras07 committed:
Fix update command and some related stuff.

kporras07 committed:
Merge branch 'v3.0' of github.com:pantheon-systems/terminus into plugin-commands-v3

kporras07 committed:
Fix coding standards.

kporras07 committed:
Start updating the install command.

kporras07 committed:
Update @todo to make them searchable.

kporras07 committed:
Throw exception if variable is different than array or string.

kporras07 committed:
Merge branch 'fix-process-use' of github.com:pantheon-systems/terminus into plugin-commands-v3

kporras07 committed:
Update install, uninstall and load mechanisms.

kporras07 committed:
Handle the right exception type.

kporras07 committed:
Filter search results by terminus compatible version.

kporras07 committed:
Remove stability option in install command.

kporras07 committed:
Remove PluginAutoloadDependencies class.

kporras07 committed:
Add comment about separating package name and version.

kporras07 committed:
Remove autoload fallback.

kporras07 committed:
Initial changes to the update command.

kporras07 committed:
Improve messages handling and use constant for backup command.

kporras07 committed:
Initial terminus-dependencies handling on plugin install.

namespacebrian committed:
CMS-166: Fix --keep-for parameter on backup:automatic:enable (#2169)

kporras07 committed:
Move terminus handling to parent class and implement in install, update and uninstall functions.

kporras07 committed:
Add and use backupDir function.

kporras07 committed:
Improve errorhandling for install and related commands.

kporras07 committed:
Improve errorhandling for update and uninstall, also create restoreBackup function.

kporras07 committed:
Add some @todos and remove autoload related code in plugins.

kporras07 committed:
Add new Reload command.

kporras07 committed:
Use dependencies dir for plugins autoload.

kporras07 committed:
Add @todo comments.

kporras07 committed:
Handle terminus-dependencies on plugin update or uninstall.

kporras07 committed:
Change to t3 binary.

kporras07 committed:
Update discovery mechanism and remove autoload because it already happened.

kporras07 committed:
Update install command to follow new folders structure.

kporras07 committed:
Initial changes to the update command.

kporras07 committed:
Remove backup related folders.

kporras07 committed:
Update other commands and bootstrap to use new structure.

kporras07 committed:
Load separate autoload for terminus itself.

kporras07 committed:
Work on temp dir and then override original dir.

kporras07 committed:
Fix commands to start from current plugins and dependencies status.

kporras07 committed:
Make update command work as expected and update some commands.

kporras07 committed:
Fix coding standards.

kporras07 committed:
Remove unused code and add some @todo comments.

kporras07 committed:
Remove comment about docblocks after review.

namespacebrian committed:
Clean up functional test groups (#2172)

* Clean up functional test groups

- Moved un-implemented tests to 'todo' test group
- Made function comment formatting more consistent

kporras07 committed:
Remove test for non-existing file.

kporras07 committed:
Use terminus dependencies base dir to improve testability.

kporras07 committed:
Use dependencies dir in feature context.

kporras07 committed:
Remove unused code in reload command.

kporras07 committed:
Add protected var with default value.

kporras07 committed:
Make self:plugin:list command pass.

Commit:
Run functional tests for 7.4 and 8.0 consequentially

stovak committed:
composer.lock update (#2173)

kporras07 committed:
Merge branch 'v3.0' of github.com:pantheon-systems/terminus into plugin-commands-v3

kporras07 committed:
Remove unused behat tests.

kporras07 committed:
Add initial tests for plugin manager.

kporras07 committed:
Add .envrc to .gitignore.

kporras07 committed:
Add remaining tests and reorganize some functions and env vars.

kporras07 committed:
Revert "Remove test for non-existing file."

This reverts commit 04b32c48eb0ed61dcd416f8c5cf2004bd92d72eb.

kporras07 committed:
Undo some test related not needed changes.

kporras07 committed:
Undo more tests related changes.

kporras07 committed:
Add Brewfile.lock.json to gitignore.

kporras07 committed:
Document terminus dependencies version and make it optional in the factory function.

kporras07 committed:
Fix coding standards.

kporras07 committed:
Update testing approach to reduce usage of env vars.

kporras07 committed:
Fix latest and stable version methods.

kporras07 committed:
Remove empty line.

Commit:
v3.0 drush:aliases command fixes and functional tests (#2175)

* Add helper methods to get test site framework

* Fix drush:aliases --print

* Fix drush:aliases (call without params)

* Add test scenario for "drush:aliases" command to verify exported Drush8 aliases

* Fix drush:aliases for Drush 9

* Use centralized Twig template processing in Template::process

* Add test scenario for "drush:aliases" command to verify exported Drush9 site alias

kporras07 committed:
Merge branch 'v3.0' of github.com:pantheon-systems/terminus into plugin-commands-v3

kporras07 committed:
Use symfony/filesystem to remove directories instead of new removeDir function.

kporras07 committed:
Make dependencies version a global constant.

kporras07 committed:
Plugin install should run composer update only for the installed plugin.

kporras07 committed:
Update windows detection mechanism.

kporras07 committed:
Remove unneeded lines.

kporras07 committed:
Use parent runComposerUpdate function from base class.

kporras07 committed:
Support t2 plugins in t3.

kporras07 committed:
Prefer stable for composer.json files.

kporras07 committed:
Support package installation without prefix.

kporras07 committed:
Fix plugins dir in env vars.

kporras07 committed:
Update changelog.

kporras07 committed:
Fix coding standards.

kporras07 committed:
Update tests invoking terminus to avoid grep and redirecting stderr in calling functions.

kporras07 committed:
Reformat changelog entry.

kporras07 committed:
Update tests to use terminus plugin example.

kporras07 committed:
Try using constant from global namespace.

kporras07 committed:
Make constant not needed in Terminus factory function.

kporras07 committed:
Validate if dependencies_version exists.

kporras07 committed:
Go back to use variable instead of constant.

kporras07 committed:
Merge pull request #2176 from pantheon-systems/plugin-commands-v3

Terminus plugins manager

greg-1-anderson commented:
3.0 development continues on the 3.x branch.
